x/claim/wasm/encoder.go

```diff
@@ -0,0 +1,63 @@
+package wasm
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	sgwasm "github.com/public-awesome/stargaze/v3/internal/wasm"
+	claimtypes "github.com/public-awesome/stargaze/v3/x/claim/types"
+)
+
+var _ sgwasm.Encoder = Encoder
+
+type ClaimAction string
+
+const (
+	ClaimActionMintNFT = "mint_nft"
+	ClaimActionBidNFT  = "bid_nft"
+)
+
+type ClaimFor struct {
+	Address string      `json:"address"`
+	Action  ClaimAction `json:"action"`
+}
+
+func (a ClaimAction) ToAction() (claimtypes.Action, error) {
+	if a == ClaimActionMintNFT {
+		return claimtypes.ActionMintNFT, nil
+	}
+
+	// rebranding the action
+	if a == ClaimActionBidNFT {
+		return claimtypes.ActionBuySocialToken, nil
```

Comment:
Okay to leave this as "social token" since its just an internal detail, but is it possible to change since this is an upgrade?

Reply:
I guess if the assigned protobuf code doesn't change it will be the same and can change it